src/runtime_src/core/edge/drm/zocl/zocl_ctrl_ert.c

```diff
@@ -402,9 +416,8 @@ static void zert_assign_cu_xgqs(struct zocl_ctrl_ert *zert)
 		return;
 	}
 
-	for (i = 0; i < zert->zce_num_cus; i++, cu++) {
-		struct platform_device *xgqpdev = NULL;
-
+	if (cu_domain == DOMAIN_PL) {
```

Comment:
Not your code @saifuddin-xilinx , but the duplicated code in this if-else (and many other places that cu and scu related) hurts my eyes badly... sigh...